api/fixtures_test.go

```diff
@@ -0,0 +1,1139 @@
+	accounts := generated.AccountsResponse{}
+	errStr = parseForProver(resp, &accounts)
+	if errStr != nil {
+		return
+	}
+	wit = struct {
+		Type     string                     `json:"goType"`
+		Accounts generated.AccountsResponse `json:"accounts"`
+	}{
+		Type:     fmt.Sprintf("%T", accounts),
+		Accounts: accounts,
+	}
+	return
+}
+func accountInfoProof(resp responseInfo) (wit interface{}, errStr *string) {
+	account := generated.AccountResponse{}
+	errStr = parseForProver(resp, &account)
+	if errStr != nil {
+		return
+	}
+	wit = struct {
+		Type    string                    `json:"goType"`
+		Account generated.AccountResponse `json:"account"`
+	}{
+		Type:    fmt.Sprintf("%T", account),
+		Account: account,
+	}
+	return
+}
+
+func appsProof(resp responseInfo) (wit interface{}, errStr *string) {
+	apps := generated.ApplicationsResponse{}
+	errStr = parseForProver(resp, &apps)
+	if errStr != nil {
+		return
+	}
+	wit = struct {
+		Type string                         `json:"goType"`
+		Apps generated.ApplicationsResponse `json:"apps"`
+	}{
+		Type: fmt.Sprintf("%T", apps),
+		Apps: apps,
+	}
+	return
+}
+
+func appInfoProof(resp responseInfo) (wit interface{}, errStr *string) {
+	app := generated.ApplicationResponse{}
+	errStr = parseForProver(resp, &app)
+	if errStr != nil {
+		return
+	}
+	wit = struct {
+		Type string                        `json:"goType"`
+		App  generated.ApplicationResponse `json:"app"`
+	}{
+		Type: fmt.Sprintf("%T", app),
+		App:  app,
+	}
+	return
+}
+
+func boxProof(resp responseInfo) (wit interface{}, errStr *string) {
+	box := generated.BoxResponse{}
+	errStr = parseForProver(resp, &box)
+	if errStr != nil {
+		return
+	}
+	wit = struct {
+		Type string                `json:"goType"`
+		Box  generated.BoxResponse `json:"box"`
+	}{
+		Type: fmt.Sprintf("%T", box),
+		Box:  box,
+	}
+	return
+}
+
+func boxesProof(resp responseInfo) (wit interface{}, errStr *string) {
+	boxes := generated.BoxesResponse{}
+	errStr = parseForProver(resp, &boxes)
+	if errStr != nil {
+		return
+	}
+	wit = struct {
+		Type  string                  `json:"goType"`
+		Boxes generated.BoxesResponse `json:"boxes"`
+	}{
+		Type:  fmt.Sprintf("%T", boxes),
+		Boxes: boxes,
+	}
+	return
+}
+
+func parseForProver(resp responseInfo, reconstructed interface{}) (errStr *string) {
+	if resp.StatusCode >= 300 {
+		s := fmt.Sprintf("%d error", resp.StatusCode)
+		errStr = &s
+		return
+	}
+	err := json.Unmarshal([]byte(resp.Body), reconstructed)
+	if err != nil {
+		s := fmt.Sprintf("unmarshal err: %s", err)
+		errStr = &s
+		return
+	}
+	return nil
+}
+
+// ---- END provers / witness generators ---- //
+
+func (f *testCase) proverFromEndoint() (prover, error) {
+	path := f.Request.Path
+	if len(path) == 0 || path[0] != '/' {
+		return nil, fmt.Errorf("invalid endpoint [%s]", path)
+	}
+	_, p, e := getProof(path[1:])
+	return p, e
+}
+
+type proofPath struct {
+	parts map[string]proofPath
+	proof prover
+}
+
+var proverRoutes = proofPath{
```

Comment:
Currently, only the following 7 endpoints are supported. But it would be easy to add others.
```
v2/accounts
v2/accounts/:account-id
v2/applications
v2/applications/:application-id
v2/applications/:application-id/box
v2/applications/:application-id/boxes
```